docs/architecture/push-notifications/mobile.md

```diff
@@ -0,0 +1,279 @@
+If the current token assigned to the device differs from the user's token, we call
+`OnRegisteredSync()` on the `PushNotificationListenerService` to:
+
+- Register the new `PushRegisteredToken` through the Bitwarden API for the active user
+- Set the `PushCurrentToken` to the new value for the active user
+
+As we can see, the `RegisterSync()` handles registering the push token _for the active user_ only.
+In the case that there are multiple users on the device, only the user active when FCM issues a new
+token will get the update through the process described thus far.
+
+For other users on the device, `RegisterSync()` is initiated when they next log in to the
+application or the application is switched to their account. This is done in the initialization of
+the `GroupingsPage`. The same comparison is done for this user, and in this case the
+`PushRegisteredToken` is still different than the `PushCurrentToken` for that user (as we've only
+updated the `PushCurrentToken` for the initial user thus far). It is at this point that the
+Bitwarden API is notified that the subsequent users are registered for the new token.
+
+#### iOS
+
+On iOS devices, push token registration occurs through the Apple Push Notification service (APNs).
+
+When a user logs in to the iOS application or switches accounts, the application loads the
+`GroupingsPage`. In the `GroupingsPage` initialization, we first check to make sure the device has
+accepted push notifications. If not, the Bitwarden push notification prompt is shown. This prompt
+explains why iOS will be requesting push notifications for the Bitwarden mobile app.
+
+If the user accepts this prompt, or if they already have accepted it, the application checks to see
+if the _current user_ has registered for push notifications within the last day. If they have never
+registered before, or if more than one day has elapsed, the Bitwarden app registers with iOS for
+push notifications, requesting a push token. This is done in the `RegisterAsync()` method in
+`iOSPushNotificationService`. `RegisterAsync()` executes the iOS platform-specific method required
+to begin the token request process from APNs.
+
+The response from APNs with the push token is received asynchronously. When a token is obtained for
+the device, the `OnRegisteredSuccess()` method in `iOSPushNotificationHandler` is triggered. This
+then calls the `OnRegisteredAsync()` method of the `PushNotificationListenerService`, passing along
+the newly-acquired token. This method is responsible for sending the push token to the back-end API
+to register the device + user combination for push notifications.
+
+:::note
+
+We are registering for a push token once a day for each account on the device. However, it is quite
+likely that the token received from iOS will be the same each day. A different token should only be
+generated for a device when the device is uninstalled and re-installed, but we check on a daily
+basis to ensure our database is up to date.
```

Comment:
There are other cases like the ones mentioned [here](https://stackoverflow.com/a/40169657) where the token is re-generated. Maybe we could put it more generic:
"A different token can be generated for a device at some specific scenarios like restoring device from a backup and even though it may happen it's not usual. However, we check on a daily
basis to ensure our database is up to date."
What do you think?

Reply:
That's a great catch!  I updated it to be:  "A different token is generated for a device in some specific scenarios, like restoring device from a backup. Even though this is not usually the case, we check on a daily basis to ensure that the Bitwarden Azure Notification Hub is up to date."

How does that sound?

Reply:
That's great 😄